x/delegation/keeper/msg_server_withdraw_rewards_test.go

```diff
@@ -211,22 +215,290 @@ var _ = Describe("msg_server_withdraw_rewards.go", Ordered, func() {
+		)
+		PayoutRewards(s, i.ALICE, payoutCoins1)
+
+		s.RunTxDelegatorSuccess(&types.MsgDelegate{
+			Creator: i.DUMMY[2],
+			Staker:  i.ALICE,
+			Amount:  3 * i.KYVE,
+		})
+
+		payoutCoins2 := sdk.NewCoins(
+			sdk.NewCoin("bcoin", math.NewInt(8*1_000_000)),
+			sdk.NewCoin("ccoin", math.NewInt(7*1_000_000)),
+		)
+		PayoutRewards(s, i.ALICE, payoutCoins2)
+
+		// ASSERT
+		delegationModuleBalanceAfter := s.GetCoinsFromModule(types.ModuleName)
+		poolsModuleBalanceAfter := s.GetCoinsFromModule(pooltypes.ModuleName)
+
+		Expect(delegationModuleBalanceAfter).To(Equal(
+			delegationModuleBalanceBefore.Add(payoutCoins1.Add(payoutCoins2...).Add(sdk.NewInt64Coin(globalTypes.Denom, int64(3*i.KYVE)))...)),
+		)
+		Expect(poolsModuleBalanceAfter).To(Equal(poolsModuleBalanceBefore.Sub(payoutCoins1.Add(payoutCoins2...)...)))
+
+		// Calculate outstanding rewards
+		// phase1:
+		//   reward: 10_000_000 acoin, 5_000_000 bcoin
+		//   delegation_shares: d0: 20, d1: 5
+		//   => d0: 20/25 * 10_000_000 = 8_000_000 acoin; 20/25 * 5_000_000 = 4_000_000 bcoin
+		//   => d1: 5/25 * 10_000_000 = 2_000_000 acoin; 5/25 * 5_000_000 = 1_000_000 bcoin
+		// phase2:
+		//   reward: 8_000_000 bcoin, 7_000_000 ccoin
+		//   delegation_shares: d0: 20, d1: 5 d2: 3
+		//   => d0: 20/28 * 8_000_000 = 5_714_285 bcoin; 20/28 * 7_000_000 = 5_000_000 ccoin
+		//   => d1: 5/28 * 8_000_000 = 1_428_571 bcoin; 5/28 * 7_000_000 = 1_250_000 ccoin
+		//   => d2: 3/28 * 8_000_000 = 857_142 bcoin; 3/28 * 7_000_000 = 750_000 ccoin
+		// SUM: d0: 8_000_000 acoin, 9_714_285 bcoin, 5_000_000 ccoin,
+		// SUM: d1: 2_000_000 acoin, 2_428_571 bcoin, 1_250_000 ccoin
+		// SUM: d2: 0 acoin, 857_142 bcoin, 750_000 ccoin
+		Expect(s.App().DelegationKeeper.GetOutstandingRewards(s.Ctx(), i.ALICE, i.DUMMY[0]).String()).To(Equal("8000000acoin,9714285bcoin,5000000ccoin"))
+		Expect(s.App().DelegationKeeper.GetOutstandingRewards(s.Ctx(), i.ALICE, i.DUMMY[1]).String()).To(Equal("2000000acoin,2428571bcoin,1250000ccoin"))
+		Expect(s.App().DelegationKeeper.GetOutstandingRewards(s.Ctx(), i.ALICE, i.DUMMY[2]).String()).To(Equal("857142bcoin,750000ccoin"))
+
+		s.RunTxDelegatorSuccess(&types.MsgWithdrawRewards{
+			Creator: i.DUMMY[0],
+			Staker:  i.ALICE,
+		})
+		s.RunTxDelegatorSuccess(&types.MsgWithdrawRewards{
+			Creator: i.DUMMY[1],
+			Staker:  i.ALICE,
+		})
+		s.RunTxDelegatorSuccess(&types.MsgWithdrawRewards{
+			Creator: i.DUMMY[2],
+			Staker:  i.ALICE,
+		})
+
+		Expect(s.GetCoinsFromAddress(i.DUMMY[0]).String()).To(Equal("8000000acoin,9714285bcoin,5000000ccoin,980000000000tkyve"))
+		Expect(s.GetCoinsFromAddress(i.DUMMY[1]).String()).To(Equal("2000000acoin,2428571bcoin,1250000ccoin,995000000000tkyve"))
+		Expect(s.GetCoinsFromAddress(i.DUMMY[2]).String()).To(Equal("857142bcoin,750000ccoin,997000000000tkyve"))
+		Expect(s.GetCoinsFromModule(types.ModuleName).String()).To(Equal("2bcoin,28000000000tkyve"))
+	})
+
+	It("Withdraw multiple denoms (exclusively)", func() {
```

Comment:
The test description is not that expressive. Maybe "Withdraw multiple denoms, one after the other".